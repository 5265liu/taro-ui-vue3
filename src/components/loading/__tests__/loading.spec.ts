import { mount } from '@vue/test-utils'
import AtLoading from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtLoading as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values },
  })
}

describe('AtLoading Snap', () => {
  it('should render initial AtLoading', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtLoading -- props size', () => {
    const wrapper = factory({ size: 15 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtLoading -- props color', () => {
    const wrapper = factory({ color: '#fff' })
    expect(wrapper.element).toMatchSnapshot()
  })
})
