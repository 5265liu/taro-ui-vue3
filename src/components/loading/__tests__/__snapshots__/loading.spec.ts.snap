// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtLoading Snap render AtLoading -- props color 1`] = `
<view
  class="at-loading"
>
  <view
    class="at-loading__ring"
    style="border: 1px solid #fff; border-color: #fff transparent transparent transparent;"
  />
  <view
    class="at-loading__ring"
    style="border: 1px solid #fff; border-color: #fff transparent transparent transparent;"
  />
  <view
    class="at-loading__ring"
    style="border: 1px solid #fff; border-color: #fff transparent transparent transparent;"
  />
</view>
`;

exports[`AtLoading Snap render AtLoading -- props size 1`] = `
<view
  class="at-loading"
  style="width: 0.32rem; height: 0.32rem;"
>
  <view
    class="at-loading__ring"
    style="width: 0.32rem; height: 0.32rem;"
  />
  <view
    class="at-loading__ring"
    style="width: 0.32rem; height: 0.32rem;"
  />
  <view
    class="at-loading__ring"
    style="width: 0.32rem; height: 0.32rem;"
  />
</view>
`;

exports[`AtLoading Snap render initial AtLoading 1`] = `
<view
  class="at-loading"
>
  <view
    class="at-loading__ring"
  />
  <view
    class="at-loading__ring"
  />
  <view
    class="at-loading__ring"
  />
</view>
`;
